bigquery_etl/parse_udf.py

```diff
@@ -34,32 +34,55 @@ class RawUdf:
     def from_file(filepath):
         """Read in a RawUdf from a SQL file on disk."""
         dirpath, basename = os.path.split(filepath)
+
+        with open(filepath) as f:
+            text = f.read()
+
+        sql = sqlparse.format(text, strip_comments=True)
+        statements = [s for s in sqlparse.split(sql) if s.strip()]
+
         prod_name = basename.replace(".sql", "")
-        internal_name = os.path.basename(dirpath) + "_" + prod_name
+        persistent_name = os.path.basename(dirpath) + "." + prod_name
+        temp_name = os.path.basename(dirpath) + "_" + prod_name
+        internal_name = None
+
+        definitions = []
+
+        for s in statements:
+            normalized_statement = " ".join(s.lower().split())
+            if normalized_statement.startswith("create or replace function"):
+                definitions.append(s)
+                if persistent_name in normalized_statement:
+                    internal_name = persistent_name
+
+            elif normalized_statement.startswith("create temp function"):
+                definitions.append(s)
+                if temp_name in normalized_statement:
+                    internal_name = temp_name
```

Comment:
this should use an else block to populate `tests`, so that `"create or replace function"` and `"create temp function"` don't need to be repeated below.
```suggestion
                    internal_name = temp_name
            else:
                tests.append(s)
```